src/ts/features/block-navigation-mode/index.ts

```diff
@@ -0,0 +1,368 @@
+                    state.panel = 'SIDEBAR'
+                }
+            },
+        }),
+        nmap({
+            id: 'clickSelection',
+            key: 'i',
+            label: 'Click Selection',
+            onPress: async () => {
+                const block = selectedBlock()
+                if (block) {
+                    await Mouse.leftClick(block as HTMLElement)
+                    Roam.moveCursorToStart()
+                }
+            },
+        }),
+        nmap({
+            id: 'clickSelectionAndGotoEnd',
+            key: 'a',
+            label: 'Click Selection and Go-to End of Line',
+            onPress: async () => {
+                const block = selectedBlock()
+                if (block) {
+                    await Mouse.leftClick(block as HTMLElement)
+                    Roam.moveCursorToEnd()
+                }
+            },
+        }),
+        map({
+            id: 'closeSplitPage',
+            key: 'Control+w',
+            label: 'Close Page in Side Bar',
+            updateView: false,
+            onPress: () => {
+                console.log('CLOSE')
+                const block = selectedBlock()
+                if (block) {
+                    const pageContainer = block.closest(`${Selectors.sidebarContent} > div`)
+                    const closeButton = pageContainer?.querySelector(Selectors.closeButton)
+                    if (closeButton) {
+                        Mouse.leftClick(closeButton as HTMLElement)
+                    }
+                }
+            },
+        }),
+        nmap({
+            id: 'insertBlockBefore',
+            key: 'Shift+o',
+            label: 'Insert Block Before',
+            onPress: async mode => {
+                await _jumpBlocksInFocusedPanel(mode, -1)
+                await insertBlockAfter()
+            },
+        }),
+        nmap({
+            id: 'insertBlockAfter',
+            key: 'o',
+            label: 'Insert Block After',
+            onPress: async () => {
+                await insertBlockAfter()
+            },
+        }),
+        nmap({
+            id: 'toggleFold',
+            key: 'z',
+            label: 'Toggle Fold Block',
+            onPress: async () => {
+                const block = selectedBlock()
+                if (block) {
+                    await Roam.toggleFoldBlock(block)
+                }
+            },
+        }),
+        nmap({
+            id: 'enterVisualMode',
+            key: 'Shift+v',
+            label: 'Enter Visual Mode',
+            onPress: async () => {
+                const block = selectedBlock()
+                if (block) {
+                    await Roam.selectBlock(block)
+                    clearBlockNavigationView()
+                }
+            },
+        }),
+        nmap({
+            id: 'undo',
+            key: 'u',
+            label: 'Undo',
+            onPress: async () => {
+                await Keyboard.simulateKey(90, 0, {key: 'z', metaKey: true})
```

Comment:
why do you have both `90` and `z` here? :) also please add constants for instances where you use this with number